pkg/serviceapi/handler_pods.go

```diff
@@ -1,17 +1,286 @@
 package serviceapi
 
 import (
+	"context"
+	"fmt"
+	"github.com/containers/libpod/libpod/define"
+	"github.com/docker/docker/pkg/signal"
+	"github.com/pkg/errors"
 	"net/http"
+	"strconv"
+	"syscall"
 
 	"github.com/containers/libpod/libpod"
 	"github.com/gorilla/mux"
 )
 
 func registerPodsHandlers(r *mux.Router) error {
-	r.Handle(unversionedPath("/pods/"), serviceHandler(pods))
+	r.Handle(unversionedPath("/libpod/pods/"), serviceHandler(pods))
+	r.Handle(unversionedPath("/libpod/pods/{name:..*}"), serviceHandler(podDelete)).Methods("DELETE")
+	r.Handle(unversionedPath("/libpod/pods/{name:..*}/exists"), serviceHandler(podExists))
+	r.Handle(unversionedPath("/libpod/pods/{name:..*}/kill"), serviceHandler(podKill))
+	r.Handle(unversionedPath("/libpod/pods/{name:..*}/pause"), serviceHandler(podPause))
+	r.Handle(unversionedPath("/libpod/pods/prune"), serviceHandler(podPrune))
+	r.Handle(unversionedPath("/libpod/pods/{name:..*}/unpause"), serviceHandler(podUnpause))
+	r.Handle(unversionedPath("/libpod/pods/{name:..*}/restart"), serviceHandler(podRestart))
+	r.Handle(unversionedPath("/libpod/pods/{name:..*}/start"), serviceHandler(podStart))
+	r.Handle(unversionedPath("/libpod/pods/{name:..*}/stop"), serviceHandler(podStop))
 	return nil
 }
 
 func pods(w http.ResponseWriter, r *http.Request, runtime *libpod.Runtime) {
 	http.NotFound(w, r)
 }
+
+func podStop(w http.ResponseWriter, r *http.Request, runtime *libpod.Runtime) {
+	var (
+		stopError error
+	)
+	allContainersStopped := true
+	ctx := context.Background()
+	name := mux.Vars(r)["name"]
+	pod, err := runtime.LookupPod(name)
+	if err != nil {
+		noSuchPodError(w, name, err)
+		return
+	}
+
+	// TODO we need to implement a pod.State/Status in libpod internal so libpod api
+	// users dont have to run through all containers.
+	podContainers, err := pod.AllContainers()
+	if err != nil {
+		Error(w, "Something went wrong", http.StatusInternalServerError, err)
+		return
+	}
+
+	for _, con := range podContainers {
+		containerState, err := con.State()
+		if err != nil {
+			Error(w, "Something went wrong", http.StatusInternalServerError, err)
+			return
+		}
+		if containerState == define.ContainerStateRunning {
+			allContainersStopped = false
+			break
+		}
+	}
+	if allContainersStopped {
+		alreadyStopped := errors.Errorf("pod %s is already stopped", pod.ID())
+		Error(w, "Something went wrong", http.StatusNotModified, alreadyStopped)
+		return
+	}
+
+	if len(r.Form.Get("t")) > 0 {
+		timeout, err := strconv.Atoi(r.Form.Get("t"))
+		if err != nil {
+			Error(w, "Something went wrong", http.StatusInternalServerError, err)
+			return
+		}
+		_, stopError = pod.StopWithTimeout(ctx, false, timeout)
+	} else {
+		_, stopError = pod.Stop(ctx, false)
+	}
+	if stopError != nil {
+		Error(w, "Something went wrong", http.StatusInternalServerError, err)
+		return
+	}
+	w.(ServiceWriter).WriteJSON(http.StatusOK, r)
```

Comment:
You don't want to send the request back to the client. Shouldn't this be name or ""?  Same for all methods...